### PR TITLE
Snapshot builder simplification

### PR DESCRIPTION
**Pull request by b00ris**

### CODE REVIEW

File: eth/stagedsync/stage_headers_snapshot.go [hunk 1]
```diff
@@ -24,77 +25,94 @@ func StageHeadersSnapshotGenCfg(db ethdb.RwKV, snapshotDir string) HeadersSnapsh
...
 func SpawnHeadersSnapshotGenerationStage(s *StageState, tx ethdb.RwTx, cfg HeadersSnapshotGenCfg, sm *snapshotsync.SnapshotMigrator, torrentClient *snapshotsync.Client, quit <-chan struct{}) error {
-	useExternalTx := tx != nil
-	if !useExternalTx {
-		var err error
-		tx, err = cfg.db.BeginRw(context.Background())
-		if err != nil {
-			return err
-		}
-		defer tx.Rollback()
+	//generate snapshot only on initial mode
```
Comment (AlexeyAkhunov) [inline eth/stagedsync/stage_headers_snapshot.go:28]: Detecting initial cycle by checking `tx == nil` is fragile. It is better to pass the argument `initialCycle` explicitly, like it is done in the `Headers` stage

File: eth/stagedsync/stage_headers_snapshot.go [hunk 2]
```diff
@@ -24,77 +25,94 @@ func StageHeadersSnapshotGenCfg(db ethdb.RwKV, snapshotDir string) HeadersSnapsh
...
+	if tx != nil {
+		s.Done()
+		return nil
 	}
 
-	to, err := stages.GetStageProgress(tx, stages.Headers)
+	readTX, err := cfg.db.BeginRw(context.Background())
```
Comment (AlexeyAkhunov) [inline eth/stagedsync/stage_headers_snapshot.go:34]: If this is a `readTX`, why it is opened with `BeginRW` ? You should use `BeginRo` instead

File: turbo/snapshotsync/snapshot_builder.go [hunk 3]
```diff
@@ -22,12 +21,251 @@ import (
 	"github.com/ledgerwatch/erigon/log"
 )
 
-//What number should we use?
+//maxReorgDepth max reorg depth. We should create snapshot after it
 const maxReorgDepth = 90000
```
Comment (AlexeyAkhunov) [inline turbo/snapshotsync/snapshot_builder.go:25]: There is constant called `params.FullImmutabilityThreshold` which is equal to that number

File: eth/stagedsync/stage_headers_snapshot.go (hunk 1 above)
Comment (b00ris) [reply eth/stagedsync/stage_headers_snapshot.go:28]: fixed

File: eth/stagedsync/stage_headers_snapshot.go (hunk 2 above)
Comment (b00ris) [reply eth/stagedsync/stage_headers_snapshot.go]: fixed

File: turbo/snapshotsync/snapshot_builder.go (hunk 3 above)
Comment (b00ris) [reply turbo/snapshotsync/snapshot_builder.go]: fixed

**Pull request merged**